# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/DecoderTemplate.h
# Commit: e903ac081227
# Full Hash: e903ac0812273c476426eba3462c4bb2977c4f2f
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 3
# Description:
#   Bug 1749047 - Move and deduplicate some utility functions to WebCodecUtils. r=chunmin
#   
#   They will be reused for the encoding side.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195699
# ==============================================================================

diff -r 029f5470314e -r e903ac081227 dom/media/webcodecs/DecoderTemplate.h
--- a/dom/media/webcodecs/DecoderTemplate.h	Fri Dec 22 21:39:04 2023 +0000
+++ b/dom/media/webcodecs/DecoderTemplate.h	Fri Dec 22 21:39:05 2023 +0000
@@ -18,6 +18,7 @@
 #include "mozilla/dom/WorkerRef.h"
 #include "mozilla/media/MediaUtils.h"
 #include "nsStringFwd.h"
+#include "WebCodecsUtils.h"
 
 namespace mozilla {
 
@@ -29,23 +30,6 @@
 class Promise;
 enum class CodecState : uint8_t;
 
-template <typename T>
-class MessageRequestHolder {
- public:
-  MessageRequestHolder() = default;
-  ~MessageRequestHolder() = default;
-
-  MozPromiseRequestHolder<T>& Request() { return mRequest; }
-  void Disconnect() { mRequest.Disconnect(); }
-  void Complete() { mRequest.Complete(); }
-  bool Exists() const { return mRequest.Exists(); }
-
- protected:
-  MozPromiseRequestHolder<T> mRequest;
-};
-
-enum class MessageProcessedResult { NotProcessed, Processed };
-
 template <typename DecoderType>
 class DecoderTemplate : public DOMEventTargetHelper {
   using Self = DecoderTemplate<DecoderType>;